game/hud/src/components/AnchorIndicator/index.tsx

```diff
@@ -0,0 +1,72 @@
+
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import * as React from 'react';
+import {Position} from '../../services/session/layout';
+
+export interface AnchorIndicatorProps {
+  name: string;     // widget name
+  pos: Position;    // current widget position
+  setAnchor: (name: string, pos: Position) => void;
+}
+
+export interface AnchorIndicatorState {
+}
+
+class AnchorIndicator extends React.Component<AnchorIndicatorProps, AnchorIndicatorState> {
+
+  constructor(props: AnchorIndicatorProps) {
+    super(props);
+  }
+
+  private anchorHere = (event: React.MouseEvent) => {
+    const el: HTMLDivElement = event.currentTarget as HTMLDivElement;
+    function a2n(a: string) : number {
+      switch(a) {
+        case 's': return -1;
+        case 'e': return 1;
+      }
+      return 0;
+    }
+    const pos: Position = Object.assign({}, this.props.pos);
+    let x: number = a2n(el.classList[1][7]);
+    let y: number = a2n(el.classList[2][7]);
```

Comment:
1, 2, and 7 feel like magic numbers here.  Although I haven't dug deeply enough to say for sure, I'm reading this as x and y are always the 2nd and 3rd class names, and we only care about the 8th character of each.  Am I reading that right?  How likely is it that those indexes may change?